### PR TITLE
add new article for using tmp files

### LINKED ISSUE #363: Best practice recommendations on how to use /tmp so that use tmp correctly
See this case as an example:

writing sess_* files to the /tmp directory on application endpoints
"identified your Live site as writing zero byte temporary files to our application server systems, filling it close to capacity with regard to the underlying file-systems’ “inode” capabilities."

site writing many files names with the format googlefonts###.tmp (replace ### with digits) to the server’s root /tmp directory. This is causing disk space issues for the server, and we may put your site in maintenance mode until this condition can be fixed.

It looks like this is related to a site’s theme, which appears to be based off the ReduxFramework.

There is currently a bug filed with that project documenting the issue:

https://github.com/ReduxFramework/redux-framework/issues/1383

Please try the recommendations suggested in that thread so this theme doesn’t write temp files to the server /tmp directory. The Redux maintainer suggests installing the Redux Framework plugin, or you may need to manually update redux within your theme.

**Redux Framework**
**Session files**

### PR DESCRIPTION
**Pull request by nataliejeremy**
This issue closes #363 and #401 @timani please review--thanks!

### CODE REVIEW

File: source/docs/articles/sites/using-tmp-files.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+---
+
+##Temporary File Management Across Multiple Application Containers
+
+If you’re using multiple servers to process requests, you’ll need to set up a temporary directory that is shared between all environments.
+
+To share the temporary directory between all environments, use: `sites/default/files/tmp`
+
+##DROPs  
+To manage your temporary files securely and reliably across DROPs when multiple server calls are required to process the files, add this to your code: `private://tmp`
```
Comment (bmackinney) [inline source/docs/articles/sites/using-tmp-files.md]: Unclear. add "this" doesn't say what `private://tmp` is or where to add it to the codebase. @timani

File: source/docs/articles/sites/using-tmp-files.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+category:
+- getting-started
+keywords: tmp, temp files, tmp files, temporary files, mulitiple application containers, distributed environments
+---
+
+##Temporary File Management Across Multiple Application Containers
+
+If you’re using multiple servers to process requests, you’ll need to set up a temporary directory that is shared between all environments.
+
+To share the temporary directory between all environments, use: `sites/default/files/tmp`
```
Comment (bmackinney) [inline source/docs/articles/sites/using-tmp-files.md]: For Drupal. Also let's expand this to tell to create the new directory and to tell Drupal that the tmp directory is there. 
Needs equivalent WP instructions

File: source/docs/articles/sites/using-tmp-files.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+##Create a Directory for Web Servers
+When the `tmp/` path points to the server `/tmp` directory, it is not accessible by the web server on an application server. You'll need to set the path to something like:
+
+`/srv/bindings/2976a45d0d6644caaead02e2cde9a55e/tmp`
+
+<div class="alert alert-warning" role="alert">
+<strong>Note</strong>: Remember to change the path depending on the value assigned by your server. However, that path is not accessible from other application servers. If you need a shared location that every server can access, we recommend using a temp directory within the files directory for plup, such as <code>private://tmp</code> or <code>public://tmp</code>.</div>
+
+##Troubleshooting
+###ReduxFramework/Theme Issues
```
Comment (bmackinney) [inline source/docs/articles/sites/using-tmp-files.md]: @joshkoenig Are these issues with Redux now resolved? 
@nataliejeremy if they are, we can possibly keep some of this section as a case study for similar future instances of the issue.

File: source/docs/articles/sites/using-tmp-files.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+<strong>Note</strong>: Avoid using the <code>temporary://</code> scheme. Temporary files written during a batch process will not be available for processing by other application servers, and the process will fail.</div>
+
+The filesystem settings are set at bootstrap when a request is processed by Drupal. The `public://`, `private://`, and `temporary://` are set dynamically depending on the application server that receives the `$_PRESSFLOW` settings variable.
+
+For sites with multiple DROPs, this value will change. Refresh the page and this value should change on the Live environment.
+
+##Create a Directory for Web Servers
+When the `tmp/` path points to the server `/tmp` directory, it is not accessible by the web server on an application server. You'll need to set the path to something like:
+
+`/srv/bindings/2976a45d0d6644caaead02e2cde9a55e/tmp`
```
Comment (timani) [inline source/docs/articles/sites/using-tmp-files.md]: `/srv/bindings/2976a45d0d6644caaead02e2cde9a55e/files/tmp` -> within files

File: source/docs/articles/sites/using-tmp-files.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+keywords: tmp, temp files, tmp files, temporary files, mulitiple application containers, distributed environments
+---
+
+##Temporary File Management Across Multiple Application Containers
+
+If you’re using multiple servers to process requests, you’ll need to set up a temporary directory that is shared between all environments.
+
+To share the temporary directory between all environments, use: `sites/default/files/tmp`
+
+##DROPs
```
Comment (timani) [inline source/docs/articles/sites/using-tmp-files.md]: I am not sure if we are still calling these DROPs. Not sure of the new standard convention though

File: source/docs/articles/sites/using-tmp-files.md (hunk 1 above)
Comment (timani) [inline source/docs/articles/sites/using-tmp-files.md]: That's legit. A table could be created to show where these are, streamwrapper and location

File: source/docs/articles/sites/using-tmp-files.md (hunk 5 above)
Comment (suzannealdrich) [inline source/docs/articles/sites/using-tmp-files.md]: Multiple Application Container Hosts

File: source/docs/articles/sites/using-tmp-files.md (hunk 1 above)
Comment (nataliejeremy) [inline source/docs/articles/sites/using-tmp-files.md]: @timani Can you add that info?

**Comment by nataliejeremy** [comment]
@timani One more for you to review/edit. Thanks!

**Comment by nataliejeremy** [comment]
@suzannealdrich Please edit/review this PR. Thx!

File: source/docs/articles/sites/using-tmp-files.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+Create a `tmp/` directory within the network file mount, which is common across all application containers. Use the following code when applicable to write temporary resources to shared directories:
+
+**Drupal**: `public://sites/default/files/tmp`  
+**WordPress**: `public://wp-content/uploads`
+
+This will allow for consistent execution of requests to temporary files for sites with more than one application container.
+
+## Private Temporary Files
+To manage your temporary files privately and reliably for when multiple application containers are required to process the request, add this to your code: `private://tmp`
```
Comment (rachelwhitton) [inline source/docs/articles/sites/using-tmp-files.md]: Is this Drupal specific or will this work for defining a common `tmp` dir for WP as well?

Comment (nataliejeremy) [inline source/docs/articles/sites/using-tmp-files.md]: @rachelwhitton I think we need to add in the WP alternative.

Comment (rachelwhitton) [inline source/docs/articles/sites/using-tmp-files.md]: So far I have not been able to change the default `/tmp` dir used by PHP. 

I added the following to wp-config.php but it did not work:
```
define( 'WP_TEMP_DIR' , '/wp-content/uploads/') ;
```

I added `<?php echo sys_get_temp_dir(); ?>` to `header.php` to display the tmp path.

Maybe CSE can help test/verify solutions for mapping an alternate tmp dir for WP users? I'm not having much luck.

File: source/docs/articles/sites/using-tmp-files.md [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+keywords: tmp, temp files, tmp files, temporary files, mulitiple application containers, distributed environments
+---
+Live sites on Professional plans and above have multiple [Application Containers](/docs/articles/sites/all-about-application-containers).  Pantheon's distributed system means that requests are spread between all of the available application servers, which is part of how we help the site scale. However, the `/tmp` directory on one instance is not able to access the `/tmp` resources on another application server. The solution is to create a common, or shared, directory within the file mount which is accessible by all environments.
+
+## Create a Common Directory
+
+Create a `tmp/` directory within the network file mount, which is common across all application containers. Use the following code when applicable to write temporary resources to shared directories:
+
+**Drupal**: `public://sites/default/files/tmp`  
+**WordPress**: `public://wp-content/uploads`
```
Comment (bmackinney) [inline source/docs/articles/sites/using-tmp-files.md]: `public://wp-content/uploads/tmp`  to match the pattern for Drupal.

File: source/docs/articles/sites/using-tmp-files.md (hunk 6 above)
Comment (bmackinney) [inline source/docs/articles/sites/using-tmp-files.md]: @timani this doc is hung up on this question. Please work with @rachelwhitton to resolve.

**Comment by timani** [comment]
We need a table to represent the stream wrappers per service level at the top as ultimately that is the confusion point, it's not just tmp affected.

public:// | xxx/files | xxx/sites/default/files
private:// | ... | ..
tmp:// | ... | ...

https://www.drupal.org/node/555118

We need to get a table to represent the modules as a index at top, modules can be an anchor to the section in the doc:

webform | config_var_name | issue queue link 
plupload  |  config_var_name | issue queue link 
views data export  |  config_var_name | issue queue link

**Comment by nataliejeremy** [comment]
Per @bmackinney add the following:
 public://tmp == sites/default/files/tmp

**Comment by rachelwhitton** [comment]
Closing in favor of [Unsupported Modules and Plugins](https://pantheon.io/docs/articles/sites/code/unsupported-modules-plugins/)

**Comment by sukottokun** [comment]
Is this live somewhere? @nataliejeremy

**Comment by nataliejeremy** [comment]
@sukottokun Not yet, we have this open PR https://github.com/pantheon-systems/documentation/pull/1219

**Pull request closed without merging**